runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -3974,45 +3974,193 @@ TR::Register *J9::X86::TreeEvaluator::checkcastinstanceofEvaluator(TR::Node *nod
          break;
       }
    auto clazz = TR::TreeEvaluator::getCastClassAddress(node->getChild(1));
-   if (clazz &&
-       !TR::Compiler->cls.isClassArray(cg->comp(), clazz) && // not yet optimized
-       (!cg->comp()->compileRelocatableCode() || cg->comp()->getOption(TR_UseSymbolValidationManager)) &&
-       !cg->comp()->getOption(TR_DisableInlineCheckCast)  &&
-       !cg->comp()->getOption(TR_DisableInlineInstanceOf))
+   if (isCheckCast && !clazz && (!cg->comp()->compileRelocatableCode() || cg->comp()->getOption(TR_UseSymbolValidationManager)))
```

Comment:
We may want to disable the optimization when `TR_DisableInlineCheckCast` is set.

Reply:
Yes, added the option.